src/Regridder.jl

```diff
@@ -633,5 +633,49 @@ function cgll2latlonz(field; DIR = "cgll2latlonz_dir", nlat = 360, nlon = 720, c
     return new_data, coords
 end
 
+"""
+    truncate_dataset(date0, timeStart, timeEnd, datafile)
+
+Truncates given data set to only regrid the dates that are used in the simulation
+"""
+function truncate_dataset(date0, timeStart, timeEnd, datafile, datapath)
+    date_start = date0 + Dates.Second(timeStart)
+    date_end = date0 + Dates.Second(timeStart + timeEnd)
+
+    ds = NCDataset(datafile, "r")
+    dates = ds["time"][:]
+
+    if date_start < dates[1]
```

Comment:
could you add some comments in these if/else statements explaining the logic in words please?